moved/src/move_execution/deposited.rs

```diff
@@ -1,31 +1,144 @@
 use {
     crate::{
         block::HeaderForExecution,
+        error::UserError,
         genesis::config::GenesisConfig,
         move_execution::{
-            create_move_vm, create_vm_session, eth_token,
+            create_move_vm, create_vm_session, eth_token, evm_native,
             gas::{new_gas_meter, total_gas_used},
-            Logs,
         },
-        primitives::{ToMoveAddress, B256},
+        primitives::{ToMoveAddress, ToMoveU256, B256},
         types::{
             session_id::SessionId,
             transactions::{DepositedTx, TransactionExecutionOutcome},
         },
     },
+    alloy::{hex, primitives::U256},
     aptos_table_natives::TableResolver,
     move_binary_format::errors::PartialVMError,
-    move_core_types::resolver::MoveResolver,
+    move_core_types::{
+        account_address::AccountAddress, language_storage::ModuleId, resolver::MoveResolver,
+        value::MoveTypeLayout,
+    },
     move_vm_runtime::module_traversal::{TraversalContext, TraversalStorage},
+    move_vm_types::values::Value,
 };
 
+const ADDRESS_LAYOUT: MoveTypeLayout = MoveTypeLayout::Address;
+const U256_LAYOUT: MoveTypeLayout = MoveTypeLayout::U256;
+
+// Topic identifying the event
+// ETHBridgeFinalized(address indexed from, address indexed to, uint256 amount, bytes extraData)
+const ETH_BRIDGE_FINALIZED: B256 = B256::new(hex!(
+    "31b2166ff604fc5672ea5df08a78081d2bc6d746cadce880747f3643d819e83d"
+));
+
 pub(super) fn execute_deposited_transaction(
     tx: &DepositedTx,
     tx_hash: &B256,
     state: &(impl MoveResolver<PartialVMError> + TableResolver),
     genesis_config: &GenesisConfig,
     block_header: HeaderForExecution,
 ) -> crate::Result<TransactionExecutionOutcome> {
+    #[cfg(any(feature = "test-doubles", test))]
+    if tx.data.is_empty() {
```

Comment:
Is this safe to not cause any unwanted deposit injections?

Reply:
This code literally does not exist outside of tests/test-doubles (this is how the conditional compilation works) so it should be safe.

Reply:
True. test-double was to allow testing between crates, not to open to production 👍

Reply:
I think this can cause unwanted side effects - what if I want to test the normal flow?

Reply:
Thanks for providing the rationale. It's good to be aware that we make it impossible to test one edge case.

I'd say the decision to alter a function's behavior should be made explicit from the call site e.g. by the use of a test only dependency. As the caller, I should tell the function what to do, not adapt to how the function behaves.

Not blocking it though, you got my approve.

Reply:
As discussed in the meeting today, we will leave the PR as-is for now. While we all agree this the current state is sub-optimal, modifying the tests to use the proper deposit flow can be done in a future PR (task for post-devnet launch).